opencti-platform/opencti-graphql/src/config/providers.js

```diff
@@ -213,66 +213,78 @@ for (let i = 0; i < providerKeys.length; i += 1) {
+            return samlRoles.map((a) => rolesMapper[a]).filter((r) => isNotEmptyField(r));
           };
-          providerLoginHandler({ email, name: userName, firstname, lastname, provider_metadata: { nameID, nameIDFormat } }, done, opts);
-        } else {
-          done({ message: 'Restricted access, ask your administrator' });
+          // endregion
+          // region groups mapping
+          const computeGroupsMapping = () => {
+            const attrGroups = groupAttributes.map((a) => (Array.isArray(samlAttributes[a]) ? samlAttributes[a] : [samlAttributes[a]]));
+            const samlGroups = R.flatten(attrGroups).filter((v) => isNotEmptyField(v));
+            const groupsMapping = mappedConfig.groups_management?.groups_mapping || [];
+            const groupsMapper = genConfigMapper(groupsMapping);
+            return samlGroups.map((a) => groupsMapper[a]).filter((r) => isNotEmptyField(r));
+          };
+          const groupsToAssociate = R.uniq(computeGroupsMapping().concat(computeRolesMapping()));
+          // endregion
+          // region organizations mapping
+          const isOrgaMapping = isNotEmptyField(mappedConfig.organizations_default) || isNotEmptyField(mappedConfig.organizations_management);
+          const computeOrganizationsMapping = () => {
+            const orgaDefault = mappedConfig.organizations_default ?? [];
+            const orgasMapping = mappedConfig.organizations_management?.organizations_mapping || [];
+            const orgaPath = mappedConfig.organizations_management?.organizations_path || ['organizations'];
+            const availableOrgas = R.flatten(
+              orgaPath.map((path) => {
+                const value = R.path(path.split('.'), profile) || [];
+                return Array.isArray(value) ? value : [value];
+              })
+            );
+            const orgasMapper = genConfigMapper(orgasMapping);
+            return [...orgaDefault, ...availableOrgas.map((a) => orgasMapper[a]).filter((r) => isNotEmptyField(r))];
+          };
+          const organizationsToAssociate = isOrgaMapping ? computeOrganizationsMapping() : [];
+          // endregion
+          logApp.info('[SAML] Login handler', { isGroupBaseAccess, groupsToAssociate });
+          if (!isGroupBaseAccess || groupsToAssociate.length > 0) {
+            const { nameID: email } = profile;
+            const opts = {
+              providerGroups: groupsToAssociate,
+              providerOrganizations: organizationsToAssociate,
+              autoCreateGroup: mappedConfig.auto_create_group ?? false,
+            };
+            providerLoginHandler({ email, name: userName, firstname, lastname, provider_metadata: { nameID, nameIDFormat } }, done, opts);
+          } else {
+            done({ message: 'Restricted access, ask your administrator' });
+          }
+        },
+        (profile) => {
+        // SAML Logout function
+          logApp.info(`[SAML] Logout done for ${profile}`);
```

Comment:
This is the actual fix, in new version there is an additionnal function that must be defined.